// fb-library-test-npm/esm2015/lib/components/card/card.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { UtilsService } from '../../services/utils.service';
/**
 * This component is a simplier card
 */
export class CardComponent {
    constructor(utils) {
        this.utils = utils;
        /**
         * Invated when the user click on the titlebar
         */
        this.toggle = new EventEmitter();
    }
}
CardComponent.decorators = [
    { type: Component, args: [{
                selector: 'fb-card',
                template: "<div class=\"card\">\n    <div class=\"title\" (click)=\"toggle.emit(1234)\"> {{title}} </div>\n    <div class=\"description\" *ngIf=\"opened\">\n        <ng-content></ng-content>\n        <hr>\n        {{utils.title}}\n    </div>\n</div>",
                styles: [".card{border:1px solid #000;border-radius:10px}.title{padding:10px}.description{font-style:italic;padding:10px}"]
            },] }
];
CardComponent.ctorParameters = () => [
    { type: UtilsService }
];
CardComponent.propDecorators = {
    title: [{ type: Input }],
    opened: [{ type: Input }],
    toggle: [{ type: Output }]
};
